### PR TITLE
teleport: 8.0.6 -> 8.1.0

### PR DESCRIPTION
**Pull request by ymatsiuk**
###### Motivation for this change
version bump

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ymatsiuk** [comment]
Result of `nixpkgs-review pr 154717` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>terraform-full</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>teleport</li>
    <li>terraform-providers.teleport</li>
  </ul>
</details>

**Comment by ymatsiuk** [comment]
Result of `nixpkgs-review pr 154717` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>terraform-full</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>teleport</li>
    <li>terraform-providers.teleport</li>
  </ul>
</details>

**Review by justinas — APPROVED** [review]

**Comment by ymatsiuk** [comment]
Result of `nixpkgs-review pr 154717` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>terraform-full</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>teleport</li>
    <li>terraform-providers.teleport</li>
  </ul>
</details>

**Comment by ymatsiuk** [comment]
Result of `nixpkgs-review pr 154717` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>terraform-full</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>teleport</li>
    <li>terraform-providers.teleport</li>
  </ul>
</details>

**Comment by ymatsiuk** [comment]
> Result of `nixpkgs-review pr 154717` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 1 package marked as broken and skipped:
> 2 packages failed to build:
> 
>     * teleport
> 
>     * terraform-providers.teleport

petgraph failed to build as a roletester dependency from the first time 🤷🏻

File: pkgs/servers/teleport/default.nix [hunk 1]
```diff
@@ -8,17 +8,17 @@
 , xdg-utils
 , nixosTests
 
-, withRoleTester ? true
+, withRoleTester ? !stdenv.isDarwin
```
Comment (ymatsiuk) [inline pkgs/servers/teleport/default.nix:11]: Since we skip tests on darwin we can save some build time as well. WDYT @justinas ?

Comment (justinas) [reply pkgs/servers/teleport/default.nix:11]: Role tester is not a unit test-only component (the name might have confused you). You can [read more about it here](https://goteleport.com/blog/testing-access-datalog/).

Looks like the darwin build was successful on ofborg. Do you have any additional information? I'm afraid I do not have a darwin machine to debug on.

Comment (ymatsiuk) [reply pkgs/servers/teleport/default.nix]: Thanks for the details. I'll try to reproduce the build failure and open a separate issue. (reverted the change)

**Comment by ymatsiuk** [comment]
#156716

**Pull request closed without merging**